File: Task/Verify-distribution-uniformity-Naive/Java/verify-distribution-uniformity-naive.java
import static java.lang.Math.abs;
import java.util.*;
import java.util.function.IntSupplier;

public class Test {

    static void distCheck(IntSupplier f, int nRepeats, double delta) {
        Map<Integer, Integer> counts = new HashMap<>();

        for (int i = 0; i < nRepeats; i++)
            counts.compute(f.getAsInt(), (k, v) -> v == null ? 1 : v + 1);

        double target = nRepeats / (double) counts.size();
        int deltaCount = (int) (delta / 100.0 * target);

        counts.forEach((k, v) -> {
            if (abs(target - v) >= deltaCount)
                System.out.printf("distribution potentially skewed "
                        + "for '%s': '%d'%n", k, v);
        });

        counts.keySet().stream().sorted().forEach(k
                -> System.out.printf("%d %d%n", k, counts.get(k)));
    }

    public static void main(String[] a) {
        distCheck(() -> (int) (Math.random() * 5) + 1, 1_000_000, 1);
    }
}

File: Task/Verify-distribution-uniformity-Naive/C-sharp/verify-distribution-uniformity-naive.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Test
{
    static void DistCheck(Func<int> func, int nRepeats, double delta)
    {
        var counts = new Dictionary<int, int>();

        for (int i = 0; i < nRepeats; i++)
        {
            int result = func();
            if (counts.ContainsKey(result))
                counts[result]++;
            else
                counts[result] = 1;
        }

        double target = nRepeats / (double)counts.Count;
        int deltaCount = (int)(delta / 100.0 * target);

        foreach (var kvp in counts)
        {
            if (Math.Abs(target - kvp.Value) >= deltaCount)
                Console.WriteLine("distribution potentially skewed for '{0}': '{1}'", kvp.Key, kvp.Value);
        }

        foreach (var key in counts.Keys.OrderBy(k => k))
        {
            Console.WriteLine("{0} {1}", key, counts[key]);
        }
    }

    public static void Main(string[] args)
    {
        DistCheck(() => new Random().Next(1, 6), 1_000_000, 1);
    }
}
